src/language-js/printer-estree.js

```diff
@@ -1808,7 +1808,11 @@ function printPathNoParens(path, options, print, args) {
         let res;
         if (isStringLiteral(n.value)) {
           const value = rawText(n.value);
-          res = '"' + value.slice(1, -1).replace(/"/g, "&quot;") + '"';
+          if (options.singleQuote === "all") {
+            res = "'" + value.slice(1, -1).replace(/'/g, "&#39;") + "'";
```

Comment:
Anything wrong with using `&apos;`? I'm fairly sure browsers that don't support `&apos;` can't run React either.

Reply:
- [ ] hmm nope, didn't know `&apos;` was a thing, will update

Reply:
It's kinda hard to find things about `&apos;`; most places talk about how it's not in the HTML4 entity list so should be avoided, but React only works on HTML5.